### DISCUSSION TITLE
SD Upload problem

### DISCUSSION THREAD

**Discussion by j0hn1k** [luc-github/ESP3D#833] [F.A.Q] [answered]
need help, board SKR3EZ screen 128x64, marlin 2.1.1, esp32e firmware and webui 2.1.1 available files on the map are displayed, everything works except for the front files on SD, error 3 uplod file failed, and heating starts (print start) from this file, although no one started printing
I tried different cards and 4 or less and more, browsers too, everything is formatted, file names are short

**Comment by j0hn1k** [comment]
![image](https://user-images.githubusercontent.com/29076185/200706985-7e0a0f55-b291-4be3-8fe2-9cbce7e836f1.png)

**Comment by luc-github** [comment, marked as answer]
the error 3 : `#define ESP_ERROR_FILE_WRITE 3` means for some reason it cannot write to SD - the Marlin Message shows:
`Deletion failed (read-only), File: gaika.gco` and it says Read only , it is showed when https://github.com/MarlinFirmware/Marlin/blob/bugfix-2.1.x/Marlin/Configuration_adv.h#L1521

`#define SDCARD_READONLY` is enabled so check your configuration_adv.h

**Comment by j0hn1k** [comment]
closed #define SDCARD_READONLY
thank you, the download seems to have begun

**Reply by luc-github** [reply]
that said you may encounter issue issue after like https://github.com/luc-github/ESP3D/discussions/830 and https://github.com/luc-github/ESP3D/issues/826

**Comment by j0hn1k** [comment]
I understand correctly that the speed was higher
//#define BINARY_FILE_TRANSFER must be closed
#define DISABLE_SERIAL_CHECKSUM must be open
Will the speed increase if you raise the baudrate?

**Reply by luc-github** [reply]
not really https://github.com/luc-github/ESP3D/discussions/535

**Comment by j0hn1k** [comment]
#define DISABLE_SERIAL_CHECKSUM tried, transfer does not work with it
//#define BINARY_FILE_TRANSFER closed
baudrate 500k
file 1Mb - 11m30s is this normal?

wi fi Sd I have from BTT but it doesn't work with SKR3

**Reply by luc-github** [reply]
>file 1Mb - 11m30s is this normal?

Yes did you read the link I gave you ? #535

**Comment by j0hn1k** [comment]
Yes I have read
I'm not very good at English, so I'm sorry, I may have missed some details

is it possible to flash BIGTREETECH TF Cloud so that it can work with SDIO or change this module to FYSETC SD-WIFI for example?

**Reply by luc-github** [reply]
no : https://github.com/luc-github/ESP3D/issues/454#issuecomment-727014388
I do not support this hack

**Comment by j0hn1k** [comment]
Thank you, you helped me a lot!

**Reply by luc-github** [reply]
sorry to not be able to do more, I have stop to support this feature because it is not well supported and speed make it almost useless
In futur if I can implement https://github.com/luc-github/ESP3D/issues/575 it should be solved and hope have robust solution

**Comment by j0hn1k** [comment]
Hello, in the Marlin firmware, when the printer head moves very quickly along a radius of 200+, micro-sticking occurs, almost invisible to the eye, but if you look closely at the finished part, you can see them on it, the STM32F7 board with a fairly fast controller, the problem seems to be in gcode reading speed. how to maximize it? if there are any recommendations what to tweak in the config file?

**Reply by luc-github** [reply]
are you sure you are asking in right place? looks like Marlin question, no?

**Reply by j0hn1k** [reply]
I'm sorry, I thought you understood this

**Reply by luc-github** [reply]
sorry I am not Marlin expert, such question is more for expert

**Reply by j0hn1k** [reply]
ок
